torchao/quantization/quant_api.py

```diff
@@ -214,3 +221,64 @@ def replace_conv2d_1x1(conv):
     _replace_with_custom_fn_if_matches_filter(
         model, replace_conv2d_1x1, filter_fn=filter_fn
     )
+
+
+def _register_parametrizations(model, subclass_constructors, filter_fn=None):
+    def insert_parametrization(lin):
+        for subclass_constructor in subclass_constructors:
+            _, args = lin.weight.__tensor_flatten__()
+            parametrize.register_parametrization(lin, "weight", subclass_constructor(*args))
+
+    _replace_with_custom_fn_if_matches_filter(
+        model,
+        insert_parametrization,
+        _is_linear if filter_fn is None else filter_fn,
+    )
+
+    return model
+
+def _get_linear_subclass_inserter(constructor, **kwargs):
```

Comment:
Why is this needed?

Reply:
some slight differences, we'll deprecate the old `_get_subclass_inserter`